ansible/roles/macrobench/tasks/main.yml

```diff
@@ -13,6 +13,7 @@
 - name: Install arewefastyet
   become: yes
   become_user: root
+  when: arewefastyet_last_exec_is_same is undefined
```

Comment:
This is my first time looking at this, could you explain what this does in layman's terms 😄

Reply:
Sure! On the snippet you highlighted we are checking if the last execution/benchmark _**is not**_ the same (same commit, same source/trigger and same workload). If the condition is true (last benchmark is different than the current one) we would execute the whole `Install arewefastyet` instruction.

The keyword `arewefastyet_last_exec_is_same` is set by the `exec` package when [preparing Ansible](https://github.com/vitessio/arewefastyet/blob/5053153cdaeb44537f44b4f2e32a65e43110eb99/go/exec/exec.go#L372) as an "[extra vars](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#defining-variables-at-runtime)". On the snippet below, if the last execution/benchmark **_is_** the same, then we set the "extra vars" `arewefastyet_last_exec_is_same` and then it will not longer be `undefined` when running Ansible and thus the whole `Install arewefastyet` step can be skipped.

```go
        if e.PreviousBenchmarkIsTheSame {
                e.AnsibleConfig.AddExtraVar(ansible.KeyLastIsSame, true)
        }
```

This whole thing is just to optimize how we run Ansible and avoid unnecessary steps, reducing the time needed to setup and cleanup the host machine from 10+ minutes to only 1-2 minutes.

Reply:
I hope that makes sense, let me know if it is unclear :)